ingest/src/test/java/com/microsoft/azure/kusto/ingest/ResourceManagerTimerTest.java

```diff
@@ -41,16 +42,58 @@ class BooleanHolder {
         });
 
         ResourceManager resourceManager = new ResourceManager(mockedClient, 1000L, 500L, null);
+        assertFalse(resourceManager.refreshIngestionResourcesTask.isRefreshedAtLeastOnce());
+        assertFalse(resourceManager.refreshIngestionAuthTokenTask.isRefreshedAtLeastOnce());
+
         int runtime = 0;
         while (!booleanHolder.gotHere && runtime < 5000) {
             Thread.sleep(100);
```

Comment:
Not the biggest fan of calling `Thread.sleep()`. TODO for a future enhancement...